torch/csrc/distributed/c10d/ProcessGroup.cpp

```diff
@@ -89,6 +89,8 @@ c10::intrusive_ptr<Backend> ProcessGroup::getBackend(
     c10::DeviceType deviceType) {
   // If there is a backend associated with this device type then return it
   if (deviceTypeToBackend_.find(deviceType) != deviceTypeToBackend_.end()) {
+    LOG(ERROR) << "Address of return: " << &deviceTypeToBackend_[deviceType]
```

Comment:
Somehow for the same rank and same pg, I do see different address here. Is this expected? @H-Huang @kwen2501

Reply:
Do you mean for the same device, if so that would be strange? Or is it just different addresses because you are passing different devices.

Reply:
OK, the finding is that, we indeed return the same address here but somehow the pybind makes it not same when calling id(var) in python, if we print(var) directly it shows same address as well.